### PR TITLE
Added TODOs

### PR DESCRIPTION
**Pull request by KrishnaKanth1729**

### CODE REVIEW

File: parser/lexer/lexer.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def lex_string(self) -> str:
             # TODO: Raise error on finding newline/EOF
             if current == quote:
                 return string
+            elif current == '\n' or current == 'EOF':
```
Comment (classPythonAddike) [inline parser/lexer/lexer.py:56]: Thanks for fixing this tho, I completely forgot about it.

Comment (classPythonAddike) [reply parser/lexer/lexer.py:56]: Can you add an exclamation mark to the end of the error?

Comment (classPythonAddike) [reply parser/lexer/lexer.py:56]: And please create a new error for this, instead of using python's `Exceptions`

File: parser/errors/errors.py [hunk 2]
```diff
@@ -90,3 +90,9 @@ class FArgsError(FException):
     @property
     def prefix(self) -> str:
         return "Error encountered - Incorrect number of arguments provided!"
+
+
+class FNameError(FException):
```
Comment (classPythonAddike) [inline parser/errors/errors.py:95]: Can you rename the error to `FKeywordError`? And change the error message to `Error encountered - Unexpected identifier!`

File: parser/parser/parse.py [hunk 3]
```diff
@@ -74,6 +74,9 @@ def parse(self, tokens: Union[List, Token]) -> Union[Program, Expression]:
             else:
                 # Undefined keyword
                 # TODO: Proper error
+                raise FNameError(tokens[0].line,
```
Comment (classPythonAddike) [inline parser/parser/parse.py:77]: You can move the `tokens[0].line` to a new line btw :smile:

**Review by classPythonAddike — CHANGES_REQUESTED** [review]
Please make the requested changes, thanks!

File: parser/lexer/lexer.py (hunk 1 above)
Comment (KrishnaKanth1729) [reply parser/lexer/lexer.py:56]: sure :smilecat:

File: parser/errors/errors.py (hunk 2 above)
Comment (KrishnaKanth1729) [reply parser/errors/errors.py:95]: sure :smilecat:

File: parser/parser/parse.py (hunk 3 above)
Comment (KrishnaKanth1729) [reply parser/parser/parse.py:77]: alright :ugundu:

File: parser/errors/errors.py [hunk 4]
```diff
@@ -92,7 +92,12 @@ def prefix(self) -> str:
         return "Error encountered - Incorrect number of arguments provided!"
 
 
-class FNameError(FException):
+class FKeyWordError(FException):
```
Comment (classPythonAddike) [inline parser/errors/errors.py:95]: `FKeywordError`

File: parser/lexer/lexer.py [hunk 5]
```diff
@@ -54,7 +54,7 @@ def lex_string(self) -> str:
             if current == quote:
                 return string
             elif current == '\n' or current == 'EOF':
-                raise EOF("EOF while scanning string")
+                raise FEOFError('EOF while scanning string!')
```
Comment (classPythonAddike) [inline parser/lexer/lexer.py:57]: You need to provide the line, and raise the error with `.raise_error()` not `raise`

File: parser/parser/parse.py [hunk 6]
```diff
@@ -72,12 +72,10 @@ def parse(self, tokens: Union[List, Token]) -> Union[Program, Expression]:
                     tokens[0].line, *[self.parse(tok) for tok in tokens[1:]]
                 )
             else:
-                # Undefined keyword
-                # TODO: Proper error
-                raise FNameError(tokens[0].line,
-                        f"{tokens[0].value} is neither a valid keyword nor a variable"
+                raise FKeyWordError(
+                    tokens[0].line, f"`{tokens[0].value}` is neither a valid keyword nor a variable!"
```
Comment (classPythonAddike) [inline parser/parser/parse.py:76]: Different arguments on different lines please, thnx

Comment (classPythonAddike) [reply parser/parser/parse.py:76]: you don't need to raise this error with `raise`

File: parser/lexer/lexer.py (hunk 5 above)
Comment (KrishnaKanth1729) [reply parser/lexer/lexer.py:57]: ok got it

File: parser/parser/parse.py (hunk 6 above)
Comment (KrishnaKanth1729) [reply parser/parser/parse.py:76]: ok

File: parser/parser/parse.py [hunk 7]
```diff
@@ -72,8 +72,9 @@ def parse(self, tokens: Union[List, Token]) -> Union[Program, Expression]:
                     tokens[0].line, *[self.parse(tok) for tok in tokens[1:]]
                 )
             else:
-                raise FKeyWordError(
-                    tokens[0].line, f"`{tokens[0].value}` is neither a valid keyword nor a variable!"
+                FKeyWordError(
+                    tokens[0].line, 
+                    f"`{tokens[0].value}` is neither a valid keyword nor a variable!"
                     ).raise_error()
```
Comment (classPythonAddike) [inline parser/parser/parse.py:78]: Reeee indentation

File: parser/lexer/lexer.py [hunk 8]
```diff
@@ -44,7 +44,7 @@ def lex_numeric(self) -> Tuple[str, str]:
     def lex_string(self) -> str:
         quote = self.reader.current_character()
         string = ""
-
+        line = self.reader.current_line_number()
```
Comment (classPythonAddike) [inline parser/lexer/lexer.py:47]: move this to line 58 please, don't store the line in a variable over here, thnx

Comment (classPythonAddike) [reply parser/lexer/lexer.py:47]: This is still pending :eyes:

File: parser/parser/parse.py (hunk 7 above)
Comment (classPythonAddike) [reply parser/parser/parse.py:78]: Still pending :eyes:

**Review by classPythonAddike — CHANGES_REQUESTED** [review]

**Comment by classPythonAddike** [comment]
Thanks for the PR! Will merge it in an hour or so after testing locally

**Review by classPythonAddike — APPROVED** [review]

**Review by pranavbaburaj — APPROVED** [review]
LGTM

**Pull request merged**